src/API/API.js

```diff
@@ -7,66 +7,62 @@ class API {
     // can I not use this in static methods because the API wasn't constructed?
   // }
 
-  static getData = async location => {
-    let response
-    let data
-    let key = this.findRelevantData(location)
+  static getData = async (location, id) => {
+    let pathAndData = this.findRelevantPathAndData(location, id);
     try {
-      response = await fetch(`${apiHead}/${location}`)
-      data = await response.json()
+      let response = await fetch(pathAndData.path)
+      let data = await response.json()
+      return data[pathAndData.data]
     } catch (error) {
       return error
     }
-    return data[key]
-    //should this get moved into the `try` block?
   }
-  
-  static findRelevantData = location => {
-    const acceptableData = ['movies', 'movie', 'videos', 'ratings']
-    const apiPath = location.split('/') 
-    if (apiPath.length === 3 
-      && acceptableData.includes(apiPath[2])) {
-      return apiPath[2]
-    } else if (apiPath.length === 2 
-      && acceptableData.includes('movies')) {
-      return 'movie'
-    } else if (location.includes('movies')) {
-      return 'movies'
+
+  static findRelevantPathAndData = (location, id) => {
```

Comment:
This looks better. It's weird the deletions are for the branch two masters ago???